### DISCUSSION TITLE
业务重启定时器怎么办

### DISCUSSION THREAD

**Discussion by taobali32** [swoole/swoole-src#4823] [Q&A]
业务场景:
2个人语音定时器每分钟给2个人推送扣除金币消息。
这个时候如果新的代码更新后定时器就没了。

**Comment by NathanFreeman** [comment]
请具体描述一下场景，而且代码重启之后定时器代码逻辑还是在的

**Reply by taobali32** [reply]
我的业务场景,

当2个人使用腾讯im开始通话后,就开启定时器每分钟扣除1次金币并执行socket推送, 
但是比如软件运行期间，我这边代码需要变动，这个时候服务器上的新代码就没这个定时器了。 后续会造成2个人一直通话不扣除金币。
![image](https://user-images.githubusercontent.com/30717763/188264234-908fea0e-0ba0-4ed9-9a9a-13bf30608a8f.png)

**Reply by NathanFreeman** [reply]
那么你需要在重启的时候通知腾讯imi关闭所有的客户端连接，然后重新走连接和扣费流程。

**Reply by taobali32** [reply]
有没有更好的办法的。从程序角度考虑，  业务这样子做肯定不行的

**Reply by NathanFreeman** [reply]
但是你服务端重启时候，腾讯imi也会因为重启导致客户端无法与服务端通信，你还是得走一遍连接流程。
